doc/about/roadmap.md

```diff
@@ -1,80 +1,297 @@
+The aim is to make it easy to build applications that look good by default, while remaining fully customizable.
 
-### Dynamic Compilation of Models
+## Core Architecture and Scope
 
-In a bid to increase efficiency and provide more flexibility, Panel will integrate with Anywidget without having to pull in ipywidgets_bokeh. This will enable dynamic compilation of models, enhancing responsiveness, leverage the growing ecosystem built on this modern tooling and providing a more seamless connection between the frontend and backend components.
+### Reduce Core Scope and Expand the Extension Ecosystem
 
-### Community
+As Panel has grown, the scope of the core library has expanded to include a wide variety of visualization backends and integrations. While powerful, this breadth increases maintenance burden and complexity for contributors and users alike.
 
-As part of the general effort to improve our developer experience we aim to better engage with the community to encourage contributors to help improve the ecosystem. This will include items such as guided sprints with core developers, producing video material aimed at educating users about contributions, diligent tagging of issues that are suitable for first time contributors and checklists to get people started. Additionally the core team will organize hackathons, community calls and live streams.
+Going forward, Panel will focus on a **leaner core** that provides:
 
-## Chat and LLM Interfaces
+* A stable application framework
+* Layout, reactivity, and state management
+* First-class integration with Bokeh and core rendering infrastructure
 
-Enhancing the interaction with Large Language Models (LLMs) through a polished and efficient interface is a primary focus in our roadmap. The following sections detail our initiatives to take the Chat and LLM interfaces to the next level.
+More specialized or domain-specific integrations will increasingly live in **separately versioned extension packages**, allowing them to evolve independently and reduce coupling in the core codebase.
 
-### Fully Polished and Full-Featured Chat Interface for Interacting with LLMs
+This includes factoring out integrations such as:
 
-Building upon the existing functionality, we aim to introduce a fully polished and full-featured Chat interface that enables seamless interaction with LLMs. This will empower users to engage with the models in a more intuitive manner, improving accessibility and widening the application of LLMs within various domains. By enabling the visualization of complex data structures, images, and other multimedia elements, the interaction with LLMs will become more engaging and informative. This feature will make the Chat interface a versatile tool for exploring and presenting information generated by LLMs.
+* Vizzu
```

Comment:
Would that mean we should also add npm packages / CDN upload to some of these? 

This shouldn't be in the roadmap, just curious.

Reply:
Likely yes, panel-graphic-walker and panel-material-ui already publish to CDN. Though I think the goal would be to transition these to become ESM components simplifying the build complexity and Bokeh compatibility issues.